server/middlewares/Authentication.js

```diff
@@ -0,0 +1,227 @@
+      .then((role) => {
+        if (role.title === 'admin') {
+          next();
+        }
+        return response.status(403)
+          .send({
+            message: 'You are not permitted to perform this action.'
+          });
+      });
+  },
+  /**
+   * Gets user session jwt token
+   * @param  {user} user Users' object
+   * @return {Object} userToken
+   */
+  getToken(user) {
+    const userToken = jwt.sign({
+      userId: user.id,
+      roleId: user.roleId
+    },
+      secretKey, { expiresIn: '3d' });
+    return userToken;
+  },
+  /**
+   * Verifies user login details
+   * @param  {Object} request  Request object
+   * @param  {Object} response Response object
+   * @param {Object} next Next process handler
+   * @return {Object}
+   */
+  verifyLogin(request, response, next) {
+    if (!request.body.password || !request.body.email) {
+      return response.status(400)
+        .send({
+          message: 'Please provide your email and password to login.'
+        });
+    }
+    const email = /\S+@\S+\.\S+/.test(request.body.email),
+      password = /\w+/g.test(request.body.password);
+    if (!email || !password) {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid email and password.'
+        });
+    }
+    next();
+  },
+  /**
+   * Checks users' input
+   * @param  {Object} request  Request object
+   * @param  {Object} response  Response object
+   * @param  {Object} next Next process handler
+   * @return {Object}
+   */
+  verifyUserInput(request, response, next) {
+    let email = /\S+@\S+\.\S+/.test(request.body.email),
+      firstName = /\w+/g.test(request.body.firstname),
+      lastName = /\w+/g.test(request.body.lastname),
+      userName = /\w+/g.test(request.body.username),
+      password = /\w+/g.test(request.body.password);
+
+    if (!email) {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid email address.'
+        });
+    }
+    if (!firstName || firstName === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid firstname.'
+        });
+    }
+    if (!lastName || lastName === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid lastname.'
+        });
+    }
+    if (!userName || userName === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid username.'
+        });
+    }
+    if (!password || password === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Password field cannot be empty. Please enter a password.'
+        });
+    }
+    if (request.body.password.length < 8) {
+      return response.status(400)
+        .send({
+          message: 'Password cannot be less than 8 characters. Try again.'
+        });
+    }
+    db.User.findOne({ where: { email: request.body.email } })
```

Comment:
why not combine the search for username and email find, and return user already exist